plugin/storage/grpc/config/config.go

```diff
@@ -90,7 +100,28 @@ func (c *Configuration) buildRemote(logger *zap.Logger, tracerProvider trace.Tra
 	opts := []grpc.DialOption{
 		grpc.WithStatsHandler(otelgrpc.NewClientHandler(otelgrpc.WithTracerProvider(tracerProvider))),
 		grpc.WithBlock(),
+		grpc.WithDefaultCallOptions(grpc.WaitForReady(c.WaitForReady)),
+	}
+
+	if c.Compression.IsCompressed() {
```

Comment:
Doesn't OTEL provide a utility function in their config packages that helps with translating gRPC server config into settings? We should avoid reinventing the wheel and just reuse what they have.

Reply:
To reuse that, we need to fully utilize all the features it has. We can do this for every feature except `auth` (authentication) , since it needs to be supported by both the client and server. and also the `auth` is tightly coupled with auth extension https://github.com/open-telemetry/opentelemetry-collector/tree/main/config/configauth

Reply:
> Do we need to do something special for auth?

yes, i think from the client side we can reuse the otel's authentication extensions but, from the server side we need to find a find a way to support authentication.